### PR TITLE
More Unit Tests

### PR DESCRIPTION
**Pull request by KimberlyMunoz**
This bumps our coverage of our utilities to about 67%.

## Additions
- Adding `type-checkers-spec.js`
- Adding `date-range-formatter-spec.js`
- Creating util unit test folder to match structure of JS

## Testing
- Run the unit tests!

## Preview

image

## Notes
- The Chai assertion `.to.be.true` kept returning eslint errors, so I've turned off the dot-notation rule for the unit testing folder

### CODE REVIEW

**Comment by imuchnik** [comment]
Nicely done! :+1:

**Comment by imuchnik** [comment]
Looks really sexy to me. I want to merge it. Can I?

**Comment by KimberlyMunoz** [comment]
I want to wait until I get one more :+1: so non-west coast folks can see it before it goes in.

**Comment by imuchnik** [comment]
Paging @contolini who is technically  a non-west-coaster now/

**Comment by contolini** [comment]
I am an all-coaster.

**Comment by contolini** [comment]
:+1:

File: test/unit_tests/modules/util/type-checkers-spec.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+var anObject = {
+  a: '1',
+  b: '2',
+  c: '3'
+};
+var anArray = [ 1, 2, 3 ];
+
+describe( 'TypeCheckers isUndefined', function() {
+  it( 'should identify undefined variables', function() {
+    expect( typeCheckers.isUndefined( undefinedVar ) ).to.be.true;
```
Comment (anselmbradford) [inline test/unit_tests/modules/util/type-checkers-spec.js]: Is there a way we could have consistent assertions between the unit tests and the browser acceptance tests? E.g. both have `to.be.true` or `toBe( true )` syntax? Preferably the latter so the ESLint `dot-notation` ignore isn't necessary. 
Not something that needs to hold up this PR, but just wondering how we might make a unified test syntax.

File: test/unit_tests/modules/util/type-checkers-spec.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
+'use strict';
+var chai = require( 'chai' );
+var expect = chai.expect;
+var typeCheckers = require(
+  '../../../../src/static/js/modules/util/type-checkers.js'
```
Comment (anselmbradford) [inline test/unit_tests/modules/util/type-checkers-spec.js]: At some point it might make sense to make a test paths module that has the JS source base path, so that this could look something like:

```
var config = require( 'config' );
var typeCheckers = require( config.jsSrcPath + 'modules/util/type-checkers.js' );
```

Perhaps could be covered in https://github.com/cfpb/cfgov-refresh/issues/861

**Comment by anselmbradford** [comment]
:+1:

File: test/unit_tests/modules/util/type-checkers-spec.js (hunk 1 above)
Comment (KimberlyMunoz) [inline test/unit_tests/modules/util/type-checkers-spec.js]: These tests are in Mocha with Chai syntax and the browser tests are in Jasmine. The only way I see us getting consistent asserts is if we switch our unit tests to Jasmine, but that seems unnecessary.

Comment (anselmbradford) [inline test/unit_tests/modules/util/type-checkers-spec.js]: Does Mocha/Chai add something that Jasmine doesn't?

Comment (KimberlyMunoz) [inline test/unit_tests/modules/util/type-checkers-spec.js]: They're both comparable. I don't know if we'd be able to use jsdom for mocking with Jasmine, but the main reason we went with Mocha is that it's what Owning a Home, eRegs and a couple other projects are using.

http://thejsguy.com/2015/01/12/jasmine-vs-mocha-chai-and-sinon.html

Comment (imuchnik) [inline test/unit_tests/modules/util/type-checkers-spec.js]: @anselmbradford @KimberlyMunoz Technically we can use [Mocha with Protractor](https://github.com/angular/protractor/blob/master/docs/frameworks.md), but it seems like a lot of work for a dubious benefit of assertion uniformity. 
TL;DR version mocha is a test framework for node, hence OAH is using it (not sure about  others), 
Jasmine is an older, more mature and more full-features framework (included spies, mocks and assertions) that is default for Protractor.
